clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -565,23 +565,10 @@ static void collectSYCLAttributes(Sema &S, FunctionDecl *FD,
                SYCLIntelNumSimdWorkItemsAttr,
                SYCLIntelSchedulerTargetFmaxMhzAttr,
                SYCLIntelMaxWorkGroupSizeAttr, SYCLIntelMaxGlobalWorkDimAttr,
-               SYCLIntelNoGlobalWorkOffsetAttr, SYCLSimdAttr>(A);
```

Comment:
I don't think this is right, @mendell27 can you comment?  It seems to me that we want the metadata to be attached to the FunctionDecl/IR Function that it is applied to, not collected and applied to callers/the kernel, right?

Reply:
Correct.  It has to be applied to the function it is attached to.

Reply:
Thanks @erichkeane and @mendell27 for the clarification.

@mendell27, does the attribute need to be emitted in host LLVM IR?
Or Does this only apply to lambdas and function definitions on device code?

Reply:
Done